lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/files/JobJarMaker.java

```diff
@@ -96,55 +102,88 @@ private static File toFile(final FileResource fileProto) {
   public File createJobSubmissionJAR(
       final JobSubmissionEvent jobSubmissionEvent,
       final Configuration driverConfiguration) throws IOException {
-    return createJAR(
-        driverConfiguration,
-        jobSubmissionEvent.getGlobalFileSet(),
-        jobSubmissionEvent.getLocalFileSet(),
-        this.fileNames.getDriverConfigurationName());
+
+    return newBuilder()
```

Comment:
prefer new JarBuilder() here to avoid calling public method inside of its class.

The benefit is that we will be able to check public methods reference count only from this class outside.

Reply:
I've never heard calling public methods inside its own class being an anti-pattern. Doesn't really matter though, so I can change it.